Functional.Primitives.AsyncEnumerables.NetStandard2/AsyncResultPartition.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+namespace Functional
+{
+	public readonly struct AsyncResultPartition<TSuccess, TFailure>
```

Comment:
This is actually just a rename. I switched to this instead of `ResultAsyncPartition` only because that matches the name for the Union version.